patches/minecraft/net/minecraft/world/level/levelgen/feature/structures/StructureTemplatePool.java.patch

```diff
@@ -1,5 +1,48 @@
 --- a/net/minecraft/world/level/levelgen/feature/structures/StructureTemplatePool.java
 +++ b/net/minecraft/world/level/levelgen/feature/structures/StructureTemplatePool.java
+@@ -41,6 +_,19 @@
+    private final ResourceLocation f_69251_;
+    private int f_69252_ = Integer.MIN_VALUE;
+ 
++   /* *************** FORGE START ************ */
++   public List<StructurePoolElement> getTemplates() {
++      List<StructurePoolElement> templatesCopy = Lists.newArrayList(f_69250_);
++      net.minecraftforge.common.ForgeHooks.modifyStructurePool(f_69248_, templatesCopy);
++      return templatesCopy;
++   }
++
++   public boolean isNotEmpty() {
++       if (!this.f_69250_.isEmpty()) return true;
++       return !this.getTemplates().isEmpty();
```

Comment:
If we replace all reference of `f_69250_` with `#getTemplates`, why would we ever check if the field itself is empty? It would only matter what `#getTemplates` returns as that is what is used.

Reply:
`getTemplates` needs to call all the modifiers, which takes time. If the list itself isn't empty, I would expect the modified list to also not be empty. Therefore, I can skip running the modifiers by checking the original list individually first.
I will admit, when writing this, I figured modifiers would be used for adding only. If someone should decide to remove things using the modifiers, this code would produce incorrect results. Considering that, I think its better to remove that line.